File: subscriber.py
from gossip_node import GossipNode
import time
import socket


def temperature(topic, content):
    print(f"Received {topic} data: {content}")


def humidity(topic, content):
    print(f"Received {topic} data: {content}")


if __name__ == "__main__":
    gossip_node = GossipNode(host='auto', port=5001)
    gossip_node.subscribe("Temperature", temperature)

    while True:
        time.sleep(1)

File: c++/subscriber.cpp
#include "GossipNode.h"
#include <iostream>
#include <chrono>
#include <thread>

int main() {
    GossipNode node("192.168.178.126", 5000);
    node.add_known_node("192.168.178.126", 5001);

    node.subscribe("Temperature", [&node](const std::string& topic, const std::string& content) {
        std::cout << "Received [" << topic << "]: " << content << std::endl;
        node.publish("Humidity", "Humidity is" +content+"%");// std::to_string(std::stoi(content))
    });

    std::cout << "Node is running... Waiting for messages." << std::endl;

    while (true) {
        std::this_thread::sleep_for(std::chrono::seconds(1));
    }

    return 0;
}
